src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -377,16 +387,24 @@ private void subscribe(String topic) {
                         .kv(LOG_KEY_REASON, subAckPacket.getReasonString())
                         .kv(LOG_KEY_TOPIC, topic)
                         .log("Successfully subscribed to topic");
+            } else if (subAckPacket.getReasonCodes().stream().allMatch(this::skipSubscribeRetry)) {
+                LOGGER.atError()
+                        .kv(LOG_KEY_REASON_CODES, subAckPacket.getReasonCodes())
+                        .kv(LOG_KEY_REASON, subAckPacket.getReasonString())
+                        .kv(LOG_KEY_TOPIC, topic)
+                        .log("Failed to subscribe to topic with a non-retryable reason code, not retrying");
             } else {
                 LOGGER.atError()
                         .kv(LOG_KEY_REASON_CODES, subAckPacket.getReasonCodes())
                         .kv(LOG_KEY_REASON, subAckPacket.getReasonString())
                         .kv(LOG_KEY_TOPIC, topic)
                         .log("Failed to subscribe to topic");
+                // throw exception to trigger RetryUtils
+                throw new CrtRuntimeException("Failed to subscribe to topic");
```

Comment:
we should throw CrtRuntimeException within our code, should just be within crt library code.  either use an messageclientexception from bridge or maybe extend it e.g. RetryableMqttOperationException